recursive_art.py

```diff
@@ -17,11 +23,38 @@ def build_random_function(min_depth, max_depth):
 
     Returns:
         The randomly generated function represented as a nested list.
-        (See the assignment writ-eup for details on the representation of
+        (See the assignment write-up for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
+    # Comment out function to exclude it from building blocks.
+    functions = [
+                ["prod", ["x"], ["y"]],
+                ["avg", ["x"], ["y"]],
+                ["cos_pi", ["x"]],
+                ["sin_pi", ["x"]],
+                ["square", ["x"]],
+                ["cube", ["x"]],
+                ["negative", ["x"]],
+                ["sum", ["x"], ["y"]],
+                ["mag", ["x"], ["y"]],
+                ["square_x_y", ["x"], ["y"]],
+                ]
+
+    depth = random.randint(min_depth, max_depth)
+
+    func = list(random.choice(functions))
+
+    if depth == 1:
+        return list(random.choice([["x"], ["y"]]))
+
+    elif depth > 1:
+        if len(func) == 2:
+            func[1] = build_random_function(depth - 1, depth - 1)
+        elif len(func) == 3:
+            func[1] = build_random_function(depth - 1, depth - 1)
+            func[2] = build_random_function(depth - 1, depth - 1)
+
+    return func
```

Comment:
Nicely written readable code. :+1:  The way you wrote this code makes it really clear what the function is doing.